connman/gweb/gweb.c

```diff
@@ -475,6 +477,30 @@ static inline void call_result_func(struct web_session *session, guint16 status)
 
 }
 
+static inline int call_dns_route_func(struct web_session *session, bool add)
+{
+	bool ret;
+
+	if (add == session->dns_routes_set)
+		return -EALREADY;
+
+	if (session->dns_route_func) {
+		ret = session->dns_route_func(session->web->index, add,
+				session->user_data);
+
+		if (!ret) {
+			debug(session->web, "Failed to %s DNS routes",
+					add ? "add" : "remove");
+			// no service or gateway, route errors do not propagate
+			return -EINVAL;
+		}
+	}
+
+	session->dns_routes_set = ret;
```

Comment:
Hmh, whoops, this `ret` does not have a value if there is no route function set.

Reply:
Better to drop that `bool ret` and use the `add` value as it is now, to set the requested value.